diff --git a/src/main.cpp b/src/main.cpp
index 34b0424..b56eaee 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -7,6 +7,10 @@
 
 size_t gAllocatedSize = 0;
 
+#include "components/renderable.hpp"
+#include "resources/shader.hpp"
+#include "resources/material.hpp"
+#include "resources/md5mesh.hpp"
 int main(int argCount, char **argValues) {
     trillek::OS os;
 #if __APPLE__
@@ -22,9 +26,15 @@ int main(int argCount, char **argValues) {
     trillek::json::System jparser;
     jparser.Parse("assets/tests/sample.json");
 
+    std::shared_ptr<trillek::graphics::Renderable> ren1(new trillek::graphics::Renderable());
+    auto mesh = trillek::resource::System::GetInstance()->Get<trillek::resource::MD5Mesh>("bob");
+    ren1->SetMesh(mesh);
+    ren1->GetMaterial()->SetShader(trillek::resource::System::GetInstance()->Get<trillek::resource::Shader>("basic_shader"));
+    ren1->UpdateBufferGroups();
 
     trillek::graphics::System gl;
     gl.Start(os.GetWindowWidth(), os.GetWindowHeight());
+    gl.AddRenderable(0, ren1);
 
     while (!os.Closing()) {
         os.OSMessageLoop();
